### PR TITLE
Added funcionality to send emails using mailhog

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Tool required for guests app.

### CODE REVIEW

File: tests/integration/config/behat.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ default:
               - admin
               - admin
             regular_user_password: 123456
+            mailhog_url: http://127.0.0.1:8025/api/v2/messages
```
Comment (PVince81) [inline tests/integration/config/behat.yml]: hard-coded port number ?

**Comment by PVince81** [comment]
Looks fine but might require pre-setting up mailhog docker in run.sh ?

File: tests/integration/config/behat.yml (hunk 1 above)
Comment (SergioBertolinSG) [inline tests/integration/config/behat.yml]: It is the default port. I don't think we want to change it? do we?

Comment (PVince81) [inline tests/integration/config/behat.yml]: ok

**Comment by SergioBertolinSG** [comment]
For using this It requires to spawn the mailhog docker container and adding this lines into config.php:

```
 'mail_domain' => 'foobar.com',
  'mail_from_address' => 'owncloud',
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => '127.0.0.1',
  'mail_smtpport' => 1025,
  'mail_smtpsecure' => '',
```

At the moment I was thinking about modifying this only in the guests app travis installation.

This is just a functionality for other apps, not sure if we want to use it also directly in core. if that is the case we should change the installation server, and jenkinsfile I guess to add the mailhog container.

**Comment by PVince81** [comment]
We will need this eventually for other things like activity email, forgotten password email, etc from core.

Regarding config.php, you should be able to change these settings with occ like you did for the skeleton. If you like you could make the test run optional (TEST_MAIL=1) and add a magic annotation for the tests that require email. But I'm not sure it's worth the effort to make this optional.

**Comment by PVince81** [comment]
:+1: ok for now.

Later on we should provide the docker directly in core.

**Comment by PVince81** [comment]
```
21:11:55   [Behat\Testwork\Argument\Exception\UnknownParameterValueException]                                              
21:11:55   Can not find a matching value for an argument `$mailhog_url` of the method `FederationContext::__construct()`.  
21:11:55                               
```

I've fixed it in my latest commit

**Comment by PVince81** [comment]
If tests pass, let's merge this.

**Pull request merged**